### ISSUE TITLE
Building wheel for bitarray (pyproject.toml) did not run successfully

### ISSUE THREAD

**Issue by bfundac** [zksync-sdk/zksync2-python#3]
`pip install zksync2`

There is a problem after using the command, which cannot be solved
Whether I install **_Visual Studio_** or download the local installation **_bitarray-2.5.1-cp38-cp38-win_amd64. whl_** in advance, it is the same error
**My python version is 3.8. How to solve this problem**

————————
![1](https://user-images.githubusercontent.com/91300069/223674391-849e4d37-8f11-4c36-ae53-428b8d74bfba.png)

> Building wheels for collected packages: bitarray, pysha3
>   Building wheel for bitarray (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Building wheel for bitarray (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [1 lines of output]
>       error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for bitarray
>   Building wheel for pysha3 (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Building wheel for pysha3 (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [9 lines of output]
>       running bdist_wheel
>       running build
>       running build_py
>       creating build
>       creating build\lib.win-amd64-cpython-38
>       copying sha3.py -> build\lib.win-amd64-cpython-38
>       running build_ext
>       building '_pysha3' extension
>       error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for pysha3
> Failed to build bitarray pysha3
> ERROR: Could not build wheels for bitarray, pysha3, which is required to install pyproject.toml-based projects

**Comment by vyastrebovvareger** [comment]
Thank you for the issue.
It has been tested under Linux based OS. I have to investigate an issue under Windows and particular env
One suggestion. Are you able to switch to the newer version of Python. 
Latest release is based on web3 py == 6.0.0

**Comment by oneprofittips** [comment]
The error remained.
`Collecting toolz>=0.8.0
  Using cached toolz-0.12.0-py3-none-any.whl (55 kB)
Building wheels for collected packages: pysha3
  Building wheel for pysha3 (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      copying sha3.py -> build\lib.win-amd64-cpython-311
      running build_ext
      building '_pysha3' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\Modules
      creating build\temp.win-amd64-cpython-311\Release\Modules\_sha3
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DPY_WITH_KECCAK=1 -ID:\XamarinProj
ects\CryptoArbitrage\PythonScripts\venv\include -IC:\Users\User\AppData\Local\Programs\Python\Python311\include -IC:\Users\User\AppData\Local\Programs\Python\Python311\Include "-IC:\Pr
ogram Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\include" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\ATLMFC\includ
e" "-IC:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Wi
ndows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt
" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /TcModules/_sha3/sha3module.c /Fobuild\temp
.win-amd64-cpython-311\Release\Modules/_sha3/sha3module.obj
      sha3module.c
      C:\Users\User\AppData\Local\Temp\pip-install-jjjxrnmi\pysha3_5da382fc1fa2429082008cfacd2c118d\Modules\_sha3\backport.inc(78): fatal error C1083: ЌҐ г¤ Ґвбп ®вЄалвм д ©« ўЄ«озҐ­ЁҐ
: pystrhex.h: No such file or directory,
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pysha3
  Running setup.py clean for pysha3
Failed to build pysha3
Installing collected packages: pywin32, pysha3, lru-dict, bitarray, websockets, urllib3, toolz, regex, pyrsistent, pycryptodome, protobuf, multidict, idna, hexbytes, frozenlist, eth-ty
ping, eth-hash, charset-normalizer, certifi, attrs, async-timeout, yarl, requests, parsimonious, jsonschema, cytoolz, aiosignal, eth-utils, aiohttp, rlp, eth-keys, eth-abi, eip712-stru
cts, eth-rlp, eth-keyfile, eth-account, web3, zksync2
  Running setup.py install for pysha3 ... error
  error: subprocess-exited-with-error

  × Running setup.py install for pysha3 did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      running install
      D:\PythonScripts\venv\Lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build 
and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      copying sha3.py -> build\lib.win-amd64-cpython-311
      running build_ext
      building '_pysha3' extension
      creating build\temp.win-amd64-cpython-311
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pysha3

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
`

**Comment by oneprofittips** [comment]
Library https://pypi.org/project/eip712-structs / has not been updated for a long time, supports the max version of python 3.7

**Comment by vyastrebovvareger** [comment]
I just checked it under Windows 10 some things

Create conda env:
conda create -n zksync2_env python=3.10

Checking ther package installation:
pip install zksync2

```
(zksync2_env) C:\Users\User>pip install zksync2
Collecting zksync2
  Downloading zksync2-0.4.0-py3-none-any.whl (41 kB)
     ---------------------------------------- 42.0/42.0 kB 2.0 MB/s eta 0:00:00
Collecting web3==6.0.0
  Downloading web3-6.0.0-py3-none-any.whl (568 kB)
     ---------------------------------------- 569.0/569.0 kB 4.5 MB/s eta 0:00:00
Collecting eip712-structs==1.1.0
  Downloading eip712-structs-1.1.0.tar.gz (12 kB)
  Preparing metadata (setup.py) ... done
Collecting eth-utils>=1.4.0
  Downloading eth_utils-2.1.0-py3-none-any.whl (24 kB)
Collecting pysha3>=1.0.2
  Downloading pysha3-1.0.2.tar.gz (829 kB)
     ---------------------------------------- 829.2/829.2 kB 10.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting lru-dict>=1.1.6
  Downloading lru_dict-1.1.8-cp310-cp310-win_amd64.whl (12 kB)
Collecting websockets>=10.0.0
  Downloading websockets-11.0.1-cp310-cp310-win_amd64.whl (124 kB)
     ---------------------------------------- 124.5/124.5 kB ? eta 0:00:00
Collecting hexbytes>=0.1.0
  Downloading hexbytes-0.3.0-py3-none-any.whl (6.4 kB)
Collecting protobuf>=4.21.6
  Downloading protobuf-4.22.1-cp310-abi3-win_amd64.whl (420 kB)
     ---------------------------------------- 420.6/420.6 kB 8.7 MB/s eta 0:00:00
Collecting eth-abi>=4.0.0-b.2
  Downloading eth_abi-4.0.0-py3-none-any.whl (28 kB)
Collecting pywin32>=223
  Downloading pywin32-306-cp310-cp310-win_amd64.whl (9.2 MB)
     ---------------------------------------- 9.2/9.2 MB 8.8 MB/s eta 0:00:00
Collecting requests>=2.16.0
  Downloading requests-2.28.2-py3-none-any.whl (62 kB)
     ---------------------------------------- 62.8/62.8 kB ? eta 0:00:00
Collecting parsimonious==0.9.0
  Downloading parsimonious-0.9.0.tar.gz (48 kB)
     ---------------------------------------- 48.7/48.7 kB ? eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting eth-typing>=3.0.0
  Downloading eth_typing-3.3.0-py3-none-any.whl (6.3 kB)
Collecting jsonschema>=4.0.0
  Downloading jsonschema-4.17.3-py3-none-any.whl (90 kB)
     ---------------------------------------- 90.4/90.4 kB ? eta 0:00:00
Collecting eth-hash[pycryptodome]>=0.5.1
  Downloading eth_hash-0.5.1-py3-none-any.whl (9.0 kB)
Collecting eth-account>=0.8.0
  Downloading eth_account-0.8.0-py3-none-any.whl (102 kB)
     ---------------------------------------- 102.4/102.4 kB ? eta 0:00:00
Collecting aiohttp>=3.7.4.post0
  Downloading aiohttp-3.8.4-cp310-cp310-win_amd64.whl (319 kB)
     ---------------------------------------- 319.8/319.8 kB 9.7 MB/s eta 0:00:00
Collecting regex>=2022.3.15
  Downloading regex-2023.3.23-cp310-cp310-win_amd64.whl (267 kB)
     ---------------------------------------- 267.9/267.9 kB 16.1 MB/s eta 0:00:00
Collecting async-timeout<5.0,>=4.0.0a3
  Downloading async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting charset-normalizer<4.0,>=2.0
  Downloading charset_normalizer-3.1.0-cp310-cp310-win_amd64.whl (97 kB)
     ---------------------------------------- 97.1/97.1 kB 5.4 MB/s eta 0:00:00
Collecting yarl<2.0,>=1.0
  Downloading yarl-1.8.2-cp310-cp310-win_amd64.whl (56 kB)
     ---------------------------------------- 56.1/56.1 kB 2.9 MB/s eta 0:00:00
Collecting attrs>=17.3.0
  Downloading attrs-22.2.0-py3-none-any.whl (60 kB)
     ---------------------------------------- 60.0/60.0 kB ? eta 0:00:00
Collecting multidict<7.0,>=4.5
  Downloading multidict-6.0.4-cp310-cp310-win_amd64.whl (28 kB)
Collecting aiosignal>=1.1.2
  Downloading aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting frozenlist>=1.1.1
  Downloading frozenlist-1.3.3-cp310-cp310-win_amd64.whl (33 kB)
Collecting eth-keys<0.5,>=0.4.0
  Downloading eth_keys-0.4.0-py3-none-any.whl (21 kB)
Collecting eth-keyfile<0.7.0,>=0.6.0
  Downloading eth_keyfile-0.6.1-py3-none-any.whl (6.5 kB)
Collecting rlp<4,>=1.0.0
  Downloading rlp-3.0.0-py2.py3-none-any.whl (20 kB)
Collecting bitarray<3,>=2.4.0
  Downloading bitarray-2.7.3-cp310-cp310-win_amd64.whl (118 kB)
     ---------------------------------------- 118.2/118.2 kB 6.7 MB/s eta 0:00:00
Collecting eth-rlp<1,>=0.3.0
  Downloading eth_rlp-0.3.0-py3-none-any.whl (5.0 kB)
Collecting pycryptodome<4,>=3.6.6
  Downloading pycryptodome-3.17-cp35-abi3-win_amd64.whl (1.7 MB)
     ---------------------------------------- 1.7/1.7 MB 9.2 MB/s eta 0:00:00
Collecting cytoolz>=0.10.1
  Downloading cytoolz-0.12.1-cp310-cp310-win_amd64.whl (321 kB)
     ---------------------------------------- 321.4/321.4 kB 6.6 MB/s eta 0:00:00
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.19.3-cp310-cp310-win_amd64.whl (62 kB)
     ---------------------------------------- 62.7/62.7 kB 3.5 MB/s eta 0:00:00
Requirement already satisfied: certifi>=2017.4.17 in d:\miniconda\envs\zksync2_env\lib\site-packages (from requests>=2.16.0->web3==6.0.0->zksync2) (2022.12.7)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
     ---------------------------------------- 140.9/140.9 kB 8.7 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ---------------------------------------- 61.5/61.5 kB 3.4 MB/s eta 0:00:00
Collecting toolz>=0.8.0
  Downloading toolz-0.12.0-py3-none-any.whl (55 kB)
     ---------------------------------------- 55.8/55.8 kB 3.0 MB/s eta 0:00:00
Building wheels for collected packages: eip712-structs, parsimonious, pysha3
  Building wheel for eip712-structs (setup.py) ... done
  Created wheel for eip712-structs: filename=eip712_structs-1.1.0-py3-none-any.whl size=11216 sha256=346070c5c05ddfc6f4961f123cab33ad3bfb1d49395f43fe1ec3e459b0006595
  Stored in directory: c:\users\user\appdata\local\pip\cache\wheels\ea\13\65\3cde864ff3ff1ec1ea7d0f76ca44ffd400be2d5e0b673ab8c2
  Building wheel for parsimonious (setup.py) ... done
  Created wheel for parsimonious: filename=parsimonious-0.9.0-py3-none-any.whl size=44378 sha256=b512208224c3accbe85f664e9502f4e632000647e2db99003c7ff4474cdffba0
  Stored in directory: c:\users\user\appdata\local\pip\cache\wheels\a8\6e\e1\5274cd97b5abf9c9fa5c7f51185d48bfc8095d8ce6ca251155
  Building wheel for pysha3 (setup.py) ... done
  Created wheel for pysha3: filename=pysha3-1.0.2-cp310-cp310-win_amd64.whl size=32918 sha256=1ce4477f18fd9f1c639f35f04fb6c0b2db1da18ba3ee034ac0f3599c8d0ce6f7
  Stored in directory: c:\users\user\appdata\local\pip\cache\wheels\23\82\72\716aca9e0ef3e10e7cda6db74bcb40c0249067b3bc6ecb6d92
Successfully built eip712-structs parsimonious pysha3
Installing collected packages: pywin32, pysha3, lru-dict, bitarray, websockets, urllib3, toolz, regex, pyrsistent, pycryptodome, protobuf, multidict, idna, hexbytes, frozenlist, eth-typing, eth-hash, charset-normalizer, attrs, async-timeout, yarl, requests, parsimonious, jsonschema, cytoolz, aiosignal, eth-utils, aiohttp, rlp, eth-keys, eth-abi, eip712-structs, eth-rlp, eth-keyfile, eth-account, web3, zksync2
Successfully installed aiohttp-3.8.4 aiosignal-1.3.1 async-timeout-4.0.2 attrs-22.2.0 bitarray-2.7.3 charset-normalizer-3.1.0 cytoolz-0.12.1 eip712-structs-1.1.0 eth-abi-4.0.0 eth-account-0.8.0 eth-hash-0.5.1 eth-keyfile-0.6.1 eth-keys-0.4.0 eth-rlp-0.3.0 eth-typing-3.3.0 eth-utils-2.1.0 frozenlist-1.3.3 hexbytes-0.3.0 idna-3.4 jsonschema-4.17.3 lru-dict-1.1.8 multidict-6.0.4 parsimonious-0.9.0 protobuf-4.22.1 pycryptodome-3.17 pyrsistent-0.19.3 pysha3-1.0.2 pywin32-306 regex-2023.3.23 requests-2.28.2 rlp-3.0.0 toolz-0.12.0 urllib3-1.26.15 web3-6.0.0 websockets-11.0.1 yarl-1.8.2 zksync2-0.4.0
```

**Comment by vyastrebovvareger** [comment]
under the building with : `python setup.py bdist_wheel build` I got also the following

```(zksync2_env) D:\zksync2>python setup.py bdist_wheel build
D:\miniconda\envs\zksync2_env\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
  warnings.warn(
D:\miniconda\envs\zksync2_env\lib\site-packages\setuptools\config\_apply_pyprojecttoml.py:103: _WouldIgnoreField: 'license' defined outside of `pyproject.toml` would be ignored.
    !!


    ##########################################################################
    # configuration would be ignored/result in error due to `pyproject.toml` #
    ##########################################################################

    The following seems to be defined outside of `pyproject.toml`:

    `license = 'MIT'`

    According to the spec (see the link below), however, setuptools CANNOT
    consider this value unless 'license' is listed as `dynamic`.

    https://packaging.python.org/en/latest/specifications/declaring-project-metadata/

    For the time being, `setuptools` will still consider the given value (as a
    **transitional** measure), but please note that future releases of setuptools will
    follow strictly the standard.

    To prevent this warning, you can list 'license' under `dynamic` or alternatively
    remove the `[project]` table from your file and rely entirely on other means of
    configuration.


!!

  warnings.warn(msg, _WouldIgnoreField)
running bdist_wheel
running build
running build_py
creating build
creating build\lib
creating build\lib\zksync2
copying zksync2\__init__.py -> build\lib\zksync2
creating build\lib\zksync2\core
copying zksync2\core\types.py -> build\lib\zksync2\core
copying zksync2\core\utils.py -> build\lib\zksync2\core
copying zksync2\core\__init__.py -> build\lib\zksync2\core
creating build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\contract_encoder_base.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\contract_factory.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\deploy_addresses.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\erc20_contract.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\eth_token.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\l1_bridge.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\l2_bridge.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\nonce_holder.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\paymaster_utils.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\precompute_contract_deployer.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\zksync_contract.py -> build\lib\zksync2\manage_contracts
copying zksync2\manage_contracts\__init__.py -> build\lib\zksync2\manage_contracts
creating build\lib\zksync2\module
copying zksync2\module\middleware.py -> build\lib\zksync2\module
copying zksync2\module\module_builder.py -> build\lib\zksync2\module
copying zksync2\module\request_types.py -> build\lib\zksync2\module
copying zksync2\module\response_types.py -> build\lib\zksync2\module
copying zksync2\module\zksync_module.py -> build\lib\zksync2\module
copying zksync2\module\zksync_provider.py -> build\lib\zksync2\module
copying zksync2\module\__init__.py -> build\lib\zksync2\module
creating build\lib\zksync2\provider
copying zksync2\provider\eth_provider.py -> build\lib\zksync2\provider
copying zksync2\provider\__init__.py -> build\lib\zksync2\provider
creating build\lib\zksync2\signer
copying zksync2\signer\eth_signer.py -> build\lib\zksync2\signer
copying zksync2\signer\__init__.py -> build\lib\zksync2\signer
creating build\lib\zksync2\transaction
copying zksync2\transaction\transaction712.py -> build\lib\zksync2\transaction
copying zksync2\transaction\transaction_builders.py -> build\lib\zksync2\transaction
copying zksync2\transaction\__init__.py -> build\lib\zksync2\transaction
creating build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\__init__.py -> build\lib\zksync2\manage_contracts\contract_abi
running egg_info
writing zksync2.egg-info\PKG-INFO
writing dependency_links to zksync2.egg-info\dependency_links.txt
writing requirements to zksync2.egg-info\requires.txt
writing top-level names to zksync2.egg-info\top_level.txt
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '__pycache__' found under directory '*'
warning: manifest_maker: MANIFEST.in, line 4: 'recursive-include' expects <dir> <pattern1> <pattern2> ...

writing manifest file 'zksync2.egg-info\SOURCES.txt'
copying zksync2\manage_contracts\contract_abi\ContractDeployer.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IERC20.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IEthToken.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IL1Bridge.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IL2Bridge.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\INonceHolder.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IPaymasterFlow.json -> build\lib\zksync2\manage_contracts\contract_abi
copying zksync2\manage_contracts\contract_abi\IZkSync.json -> build\lib\zksync2\manage_contracts\contract_abi
$(MINICONDA_PATH)\envs\zksync2_env\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
installing to build\bdist.win-amd64\wheel
running install
running install_lib
creating build\bdist.win-amd64
creating build\bdist.win-amd64\wheel
creating build\bdist.win-amd64\wheel\zksync2
creating build\bdist.win-amd64\wheel\zksync2\core
copying build\lib\zksync2\core\types.py -> build\bdist.win-amd64\wheel\.\zksync2\core
copying build\lib\zksync2\core\utils.py -> build\bdist.win-amd64\wheel\.\zksync2\core
copying build\lib\zksync2\core\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\core
creating build\bdist.win-amd64\wheel\zksync2\manage_contracts
creating build\bdist.win-amd64\wheel\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\ContractDeployer.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IERC20.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IEthToken.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IL1Bridge.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IL2Bridge.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\INonceHolder.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IPaymasterFlow.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\IZkSync.json -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_abi\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts\contract_abi
copying build\lib\zksync2\manage_contracts\contract_encoder_base.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\contract_factory.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\deploy_addresses.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\erc20_contract.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\eth_token.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\l1_bridge.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\l2_bridge.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\nonce_holder.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\paymaster_utils.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\precompute_contract_deployer.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\zksync_contract.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
copying build\lib\zksync2\manage_contracts\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\manage_contracts
creating build\bdist.win-amd64\wheel\zksync2\module
copying build\lib\zksync2\module\middleware.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\module_builder.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\request_types.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\response_types.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\zksync_module.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\zksync_provider.py -> build\bdist.win-amd64\wheel\.\zksync2\module
copying build\lib\zksync2\module\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\module
creating build\bdist.win-amd64\wheel\zksync2\provider
copying build\lib\zksync2\provider\eth_provider.py -> build\bdist.win-amd64\wheel\.\zksync2\provider
copying build\lib\zksync2\provider\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\provider
creating build\bdist.win-amd64\wheel\zksync2\signer
copying build\lib\zksync2\signer\eth_signer.py -> build\bdist.win-amd64\wheel\.\zksync2\signer
copying build\lib\zksync2\signer\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\signer
creating build\bdist.win-amd64\wheel\zksync2\transaction
copying build\lib\zksync2\transaction\transaction712.py -> build\bdist.win-amd64\wheel\.\zksync2\transaction
copying build\lib\zksync2\transaction\transaction_builders.py -> build\bdist.win-amd64\wheel\.\zksync2\transaction
copying build\lib\zksync2\transaction\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2\transaction
copying build\lib\zksync2\__init__.py -> build\bdist.win-amd64\wheel\.\zksync2
running install_egg_info
Copying zksync2.egg-info to build\bdist.win-amd64\wheel\.\zksync2-0.4.0-py3.10.egg-info
running install_scripts
creating build\bdist.win-amd64\wheel\zksync2-0.4.0.dist-info\WHEEL
creating 'dist\zksync2-0.4.0-py3-none-any.whl' and adding 'build\bdist.win-amd64\wheel' to it
adding 'zksync2/__init__.py'
adding 'zksync2/core/__init__.py'
adding 'zksync2/core/types.py'
adding 'zksync2/core/utils.py'
adding 'zksync2/manage_contracts/__init__.py'
adding 'zksync2/manage_contracts/contract_encoder_base.py'
adding 'zksync2/manage_contracts/contract_factory.py'
adding 'zksync2/manage_contracts/deploy_addresses.py'
adding 'zksync2/manage_contracts/erc20_contract.py'
adding 'zksync2/manage_contracts/eth_token.py'
adding 'zksync2/manage_contracts/l1_bridge.py'
adding 'zksync2/manage_contracts/l2_bridge.py'
adding 'zksync2/manage_contracts/nonce_holder.py'
adding 'zksync2/manage_contracts/paymaster_utils.py'
adding 'zksync2/manage_contracts/precompute_contract_deployer.py'
adding 'zksync2/manage_contracts/zksync_contract.py'
adding 'zksync2/manage_contracts/contract_abi/ContractDeployer.json'
adding 'zksync2/manage_contracts/contract_abi/IERC20.json'
adding 'zksync2/manage_contracts/contract_abi/IEthToken.json'
adding 'zksync2/manage_contracts/contract_abi/IL1Bridge.json'
adding 'zksync2/manage_contracts/contract_abi/IL2Bridge.json'
adding 'zksync2/manage_contracts/contract_abi/INonceHolder.json'
adding 'zksync2/manage_contracts/contract_abi/IPaymasterFlow.json'
adding 'zksync2/manage_contracts/contract_abi/IZkSync.json'
adding 'zksync2/manage_contracts/contract_abi/__init__.py'
adding 'zksync2/module/__init__.py'
adding 'zksync2/module/middleware.py'
adding 'zksync2/module/module_builder.py'
adding 'zksync2/module/request_types.py'
adding 'zksync2/module/response_types.py'
adding 'zksync2/module/zksync_module.py'
adding 'zksync2/module/zksync_provider.py'
adding 'zksync2/provider/__init__.py'
adding 'zksync2/provider/eth_provider.py'
adding 'zksync2/signer/__init__.py'
adding 'zksync2/signer/eth_signer.py'
adding 'zksync2/transaction/__init__.py'
adding 'zksync2/transaction/transaction712.py'
adding 'zksync2/transaction/transaction_builders.py'
adding 'zksync2-0.4.0.dist-info/METADATA'
adding 'zksync2-0.4.0.dist-info/WHEEL'
adding 'zksync2-0.4.0.dist-info/top_level.txt'
adding 'zksync2-0.4.0.dist-info/RECORD'
removing build\bdist.win-amd64\wheel
```

**Comment by 1491928274** [comment]
MAC ,python=3.10 and python=3.8, also encountered this issue.

**Comment by danijelTxFusion** [comment]
The maximum supported version of python is 3.10

**Comment by ronev** [comment]
I have the same problem on macos with python3.11 :(

**Comment by danijelTxFusion** [comment]
> I have the same problem on macos with python3.11 :(

The latest supported version is python3.10

**Comment by 1491928274** [comment]
> 我在使用 python3.11 的 macOS 上也遇到同样的问题:(

https://github.com/zksync-sdk/zksync2-python/issues/22

**Comment by 1491928274** [comment]
> I have the same problem on macos with python3.11 :(
https://github.com/zksync-sdk/zksync2-python/issues/22

**Comment by Shawnlee101** [comment]
it`s not related to the version of python. I sovled it by downloading cpp-build-tools [https://visualstudio.microsoft.com/visual-cpp-build-tools/](https://visualstudio.microsoft.com/visual-cpp-build-tools/). I think zksync2 package needs cpp tools to help compile. Download the tool, restart PC, it may work.

**Comment by BoboTiG** [comment]
Fixed by #52.
